src/javaharness/java/arcs/api/Type.java

```diff
@@ -0,0 +1,38 @@
+package arcs.api;
+
+public class Type {
+    enum Tag {
+        UNKNOWN, ENTITY, COLLECTION;
+        public boolean isCollection() {
+            return this == COLLECTION;
+        }
+        public boolean isEntity() {
+            return this == ENTITY;
+        }
+        static Tag fromString(String tag) {
+            switch (tag) {
+                case "Collection": return Tag.COLLECTION;
+                case "Entity": return Tag.ENTITY;
+                default: return UNKNOWN;
+            }
+        }
+    }
+
+    public final Tag tag;
+
+    public Type(Tag tag) {
+        this.tag = tag;
+    }
+
+    public boolean isCollection() {
```

Comment:
Will `Type` eventually do more? Otherwise we can just make `Type` itself the `Tag` enum, basically, since we're just wrapping it with two levels of the same methods currently.

Reply:
yes, I was going to replicate the way TS code handles types, so I was going to add `EntityType extends Type` and `CollectionType extends Type` (each with Entity/Collection appropriate extra fields) in the next PR.